### PR TITLE
[docs] Add link to React Native environment setup and use newer cli

### LINKED ISSUE #10521: [docs] Bare "up and running" guide is missing react native setup
We assume that React Native (CLI) is properly installed on the developer's machine in [this guide](https://docs.expo.io/bare/hello-world/). But, if that developer mainly used the managed workflow, that's not always true. Would be good to have a warning about that and a link to https://reactnative.dev/docs/environment-setup.

In one case I know the developer:
- installed the wrong Java version
- didn't have Android 9 SDK installed, causing [this "obscure error"](https://stackoverflow.com/questions/60844245/how-solve-could-not-initialize-class-org-codehaus-groovy-reflection-reflectionc)
- had [this issue with Gradle](https://stackoverflow.com/questions/60844245/how-solve-could-not-initialize-class-org-codehaus-groovy-reflection-reflectionc)

Most of these cryptic errors would have been avoided by that link or warning I think :)

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Fixes #10521

# How

Updated the docs

# Test Plan

Docs only change

### CODE REVIEW

File: docs/pages/bare/hello-world.md [hunk 1]
```diff
@@ -5,13 +5,15 @@ sidebar_label: Up and Running
...
+
+After this, let's get started with a bare project. Run `expo init` and choose one of the bare templates. We'll use the minimum templat ehere.
 
 ```sh
+# If you don't have @react-native-community/cli, install it and check out https://reactnative.dev/docs/environment-setup
+npm i -g @react-native-community/cli
 # If you don't have expo-cli yet, get it
 npm i -g expo-cli
-# If you don't have react-native-cli yet, get it
-npm i -g react-native-cli
```
Comment (byCedric) [inline docs/pages/bare/hello-world.md:14]: We (luckily) have no other references to this package in our docs 😅

File: docs/pages/bare/hello-world.md [hunk 2]
```diff
@@ -5,13 +5,15 @@ sidebar_label: Up and Running
...
 > This guide assumes that you have Xcode and/or Android Studio installed and working. It also assumes that you want to create a new project. If you have an existing app that you would like to integrate the Expo SDK in an existing app, read the [existing apps guide](/bare/existing-apps/).
 
-To get started with a bare React Native project, run `expo init` and choose one of the bare templates. We'll use the minimum template here.
+Before you get started with a bare React Native project, make sure you set up your environment for [React Native CLI](https://reactnative.dev/docs/environment-setup).
+
+After this, let's get started with a bare project. Run `expo init` and choose one of the bare templates. We'll use the minimum templat ehere.
 
 ```sh
+# If you don't have @react-native-community/cli, install it and check out https://reactnative.dev/docs/environment-setup
+npm i -g @react-native-community/cli
```
Comment (brentvatne) [inline docs/pages/bare/hello-world.md:14]: they actually recommend using `npx` these days so we can probably remove this

File: docs/pages/bare/hello-world.md [hunk 3]
```diff
@@ -5,13 +5,15 @@ sidebar_label: Up and Running
 
 > This guide assumes that you have Xcode and/or Android Studio installed and working. It also assumes that you want to create a new project. If you have an existing app that you would like to integrate the Expo SDK in an existing app, read the [existing apps guide](/bare/existing-apps/).
 
-To get started with a bare React Native project, run `expo init` and choose one of the bare templates. We'll use the minimum template here.
+Before you get started with a bare React Native project, make sure you set up your environment for [React Native CLI](https://reactnative.dev/docs/environment-setup).
+
+After this, let's get started with a bare project. Run `expo init` and choose one of the bare templates. We'll use the minimum templat ehere.
```
Comment (brentvatne) [inline docs/pages/bare/hello-world.md:10]:
```suggestion
After this, let's get started with a bare project. Run `expo init` and choose one of the bare templates. We'll use the minimum template here.
```

**Review by brentvatne — APPROVED** [review]

File: docs/pages/bare/hello-world.md (hunk 2 above)
Comment (byCedric) [reply docs/pages/bare/hello-world.md:14]: Hmm, we might want to help them setup NCC (like `create-react-native-app`). Waiting almost 30s on a fast MacBook is terrible DX :|
Screenshot 2020-10-07 at 20 13 50

**Pull request merged**